library/public/types.h

```diff
@@ -15,20 +15,38 @@ namespace f3d
  */
 struct point3_t : std::array<double, 3>
 {
+  using array::array;
+  template<typename... Args>
+  point3_t(Args&&... args)
+    : array({ double(std::forward<Args>(args))... })
+  {
+  }
```

Comment:
This makes more sense to me:

```suggestion
  using base = std::array<double, 3>;
  template<typename... Args>
  point3_t(Args&&... args)
    : base({ std::forward<Args>(args)... })
  {
  }
```

Not sure what `array::array` means.